libs/crypto/proof_of_work.go

```diff
@@ -0,0 +1,111 @@
+package crypto
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"log"
+	"math"
+	"math/big"
+)
+
+const (
+	Sha3     = "sha3_24_rounds"
+	maxNonce = math.MaxInt64
+)
+
+var prefix = []byte("Vega_SPAM_PoW")
+
+// PoW calculates proof of work given block hash, transaction hash, target difficulty and a hash function.
+// returns the nonce, the hash and th error if any.
+func PoW(blockHash string, txID string, difficulty int, hashFunction string) (int, []byte, error) {
+	var hashInt big.Int
+	var h []byte
+	var err error
+	nonce := 0
```

Comment:
Couldn't / shouldn't we use only uint(64) here?

Reply:
it is.